src/pages/home/components/dropdown.tsx

```diff
@@ -0,0 +1,50 @@
+import { classnames } from 'helpers/utils';
+import React, {
+  ReactNode, useEffect, useRef, useState,
+} from 'react';
+import styles from './dropdown.module.scss';
+
+interface DropDownProps {
+  username: string,
+  children: ReactNode,
+}
+
+const DropDown = (props: DropDownProps) => {
+  const [show, setShow] = useState(false);
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: any) => {
+      if (ref.current && !ref.current.contains(event.target) && show) {
+        setShow(false);
+      }
+    };
+    document.addEventListener('click', handleClickOutside, true);
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true);
+    };
+  }, [show]);
+
+  const handleClick = () => {
+    setShow(!show);
+  };
+
+  return (
+    <div className={styles.dropdown} ref={ref}>
+      <button
```

Comment:
you could create a Button component with the `type="button"` and then reutilize it